src/main/groovy/com/bmuschko/gradle/docker/tasks/AbstractCredentialsAwareRemoteApiTask.groovy

```diff
@@ -0,0 +1,36 @@
+package com.bmuschko.gradle.docker.tasks
+
+import com.bmuschko.gradle.docker.DockerRegistryCredentials
+import com.github.dockerjava.api.model.AuthConfig
+import groovy.transform.CompileStatic
+
+@CompileStatic
+abstract class AbstractCredentialsAwareRemoteApiTask extends AbstractDockerRemoteApiTask
```

Comment:
Let's use delegation instead of another parent class here. Can you explain why this class was introduced? Is the only reason to have the method `resolveAuthConfig` available?

Reply:
I have introduced as there are 3 classes having the same block of code, which I have added to ``resolveAuthConfig`` method. The third one is DockerBuild class, but I did not change it in scope of this PR.

I can change it to have a delegation, something like ``authConfigResolver.resolve``, but then AuthConfigResolver will need to be provided with DockerRegistryCredentials and encapsulate ``RegistryAuthLocator.groovy``. 

Maybe it is a good idea to put this method in the ``RegistryAuthLocator.groovy`` class itself. I think it is not a bad idea, just worrying that RegistryAuthLocator.groovy is getting a bit too big. What do you think?